### PR TITLE
upgrade: Upgrade compute services early (SOC-10273)

### PR DESCRIPTION
**Pull request by skazi0**
As RPC 4.x support was dropped with https://review.opendev.org/#/c/543580/
Rocky controller can't talk to Pike compute anymore. We need to upgrade
nova compute services before the controllers so that everything is running
on new RPC when we get to live-migrate the VMs during upgrade.

### CODE REVIEW

File: chef/cookbooks/crowbar/templates/default/crowbar-pre-upgrade.sh.erb [hunk 1]
```diff
@@ -86,6 +86,10 @@ log "No HA setup found..."
 
 <% if @compute_node %>
 
+log "Upgrading packages needed for nova-compute node"
+
+zypper --non-interactive up openstack-nova-compute
+
 log "Restarting services for Pike"
```
Comment (dirkmueller) [inline chef/cookbooks/crowbar/templates/default/crowbar-pre-upgrade.sh.erb]: can you fix the "Pike" comment?

**Comment by jsuchome** [comment]
So is this at the same place we had it during 6-7 upgrade?

**Comment by skazi0** [comment]
@jsuchome I have not confirmed it yet but it seems to be exactly the same situation as we had before.

File: chef/cookbooks/crowbar/templates/default/crowbar-pre-upgrade.sh.erb (hunk 1 above)
Comment (skazi0) [reply chef/cookbooks/crowbar/templates/default/crowbar-pre-upgrade.sh.erb]: done

**Pull request merged**